src/tests/Common/XUnitWrapperGenerator/XUnitWrapperGenerator.cs

```diff
@@ -118,6 +118,53 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
 
         var allTests = testsInSource.Collect().Combine(testsInReferencedAssemblies.Collect()).Combine(outOfProcessTests.Collect()).SelectMany((tests, ct) => tests.Left.Left.AddRange(tests.Left.Right).AddRange(tests.Right));
 
+        context.RegisterImplementationSourceOutput(
+            methodsInSource
+            .Combine(context.AnalyzerConfigOptionsProvider)
+            .Where(data =>
+            {
+                var (_, configOptions) = data;
+                return configOptions.GlobalOptions.InMergedTestDirectory();
+            }),
+            static (context, data) =>
+            {
+                var (method, _) = data;
+
+                // Only check test methods
+                bool found = false;
+                foreach (var attr in method.GetAttributesOnSelfAndContainingSymbols())
+                {
+                    switch (attr.AttributeClass?.ToDisplayString())
+                    {
+                        case "Xunit.ConditionalFactAttribute":
+                        case "Xunit.FactAttribute":
+                        case "Xunit.ConditionalTheoryAttribute":
+                        case "Xunit.TheoryAttribute":
+                            found = true;
+                            break;
+                    }
+                }
+                if (!found) return;
+
+                // Find methods where all returns are the literal 100 (and there is at least one return)
+                if (method.DeclaringSyntaxReferences.IsEmpty) return;
+
+                found = false;
+                foreach (SyntaxReference node in method.DeclaringSyntaxReferences.Where(n => n != null))
+                {
+                    foreach (ReturnStatementSyntax rs in node.GetSyntax(context.CancellationToken).DescendantNodes().OfType<ReturnStatementSyntax>())
+                    {
+                        if (rs.Expression == null) continue;
+                        found = true;
+                        if (rs.Expression is LiteralExpressionSyntax les && les.Token.Value is int i && i == 100) continue;
+                        return;
+                    }
+                }
+
+                if (!found) return;
```

Comment:
For a future PR, I think it's worthwhile moving these checks into a separate analyzer instead of doing them as part of the source generator. If we do them in an analyzer, we can get better performance at build time and we could use better tools (like the `IOperation` APIs) to validate these sorts of scenarios.

Reply:
I like this idea but mainly did it this way because I'm unfamiliar with all of this and couldn't easily figure out how to set anything up (primarily copying and pasting in this file). So I'll either need a template for doing so or more time to turn the docs/bing chat into something real.